data.py

```diff
@@ -0,0 +1,91 @@
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+from pathlib import Path
+import math
+import numpy as np
+import torch
+from torch.nn.utils.rnn import pad_sequence
+import utils
+
+def collate_fn(samples):
+    batch_size=len(samples)
+    strokes, texts = zip(*samples)
+
+    all_strokes = pad_sequence([torch.tensor(s) for s in strokes])
+    all_texts = pad_sequence([torch.tensor(t) for t in texts])
+    strokes_mask = torch.ones((all_strokes.shape[0], batch_size), dtype=torch.bool)
+    texts_mask = torch.ones((all_texts.shape[0], batch_size), dtype=torch.bool)
+    for i in range(batch_size):
+        this_texts_len = texts[i].shape[0]
+        texts_mask[this_texts_len:,i] = 0
+        this_stroke_len = strokes[i].shape[0]
+        strokes_mask[this_stroke_len:,i] = 0
+
+    return all_texts, texts_mask, all_strokes, strokes_mask
+    
+class HandwritingDataset(Dataset):
+    CHARSET="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789.,\" -"
```

Comment:
How did you get the charset?

Reply:
Retrieved actual chars from sample data now.